actionview/lib/action_view/helpers/asset_tag_helper.rb

```diff
@@ -362,6 +364,13 @@ def preload_link_tag(source, options = {})
         integrity = options[:integrity]
         nopush = options.delete(:nopush) || false
         rel = mime_type == "module" ? "modulepreload" : "preload"
+        add_nonce = content_security_policy_nonce &&
+          respond_to?(:request) &&
```

Comment:
Hum, not a big fan of this `respond_to?`, is this to support mailers?

Reply:
I got the idea to grab the value from `request` by how `content_security_policy_nonce` is implemented (https://github.com/rails/rails/blob/main/actionpack/lib/action_controller/metal/content_security_policy.rb#L78)

The `respond_to?` check is taken from `send_preload_links_header` further down in this file (https://github.com/rails/rails/blob/main/actionview/lib/action_view/helpers/asset_tag_helper.rb#L659) and I was assuming it's related to mailers or so. But then again calling `content_security_policy_nonce` won't do that check anyway, maybe it's only necessary because `send_preload_links_header` sends the header...?

Reply:
Yes, it's the request that hold this info so this part is correct.